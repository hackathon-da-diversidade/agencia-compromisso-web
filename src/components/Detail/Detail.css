h1 {
    font-size: 16px;
    font-weight: bold;
    margin-top: 2em;
}

span {
    padding-left: 0;
}

.Details {
    display: inline-block;
    width: 90%;
    max-width: 400px;
    text-align: left;
}