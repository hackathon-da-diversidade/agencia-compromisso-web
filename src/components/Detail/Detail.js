import React, { useState, useEffect } from 'react';
import Alert from '@material-ui/lab/Alert';
import Header from '../Header/Header';
import Button from '../UI/Button/Button';
import Information from '../UI/Information/Information';
import SocialInformation from './sections/SocialInformation';
import PersonalInformation from './sections/PersonalInformation';
import MeasuresInformation from './sections/MeasuresInformation';

import fitModelAPI from '../../api/fitModelAPI';
import classes from './Detail.module.css';
import { calculateAge } from '../../utils/dateUtils';
import { GENDER_EXPRESSION } from '../../utils/constants';

const Detail = ({ match, location }) => {
  try {
    const model = useFitModel(match.params.id);
    return (
      <div>
        <Header path="/lista" />
        <div className={classes.Details}>
          <SuccessMessage location={location} />
          <MainInfo model={model} />
          <PersonalInformation data={model} />
          <MeasuresInformation data={model.sizes} />
          <SocialInformation data={model.socialInformation} />
          <Notes notes={model.notes} />
          <ContactButton
            phoneNumber={model.phoneNumber}
            guardianPhoneNumber={model.guardianPhoneNumber}
          />
        </div>
      </div>
    );
  } catch {
    return (
      <Alert severity="warning">Não foi possível carregar o perfil.</Alert>
    );
  }
};

export default Detail;

const useFitModel = id => {
  const [model, setModel] = useState({ sizes: {}, socialInformation: {} });

  useEffect(() => {
    const loadModelInfo = async id => {
      try {
        const { data } = await fitModelAPI.get(id);
        setModel(data);
      } catch (error) {
        console.log(error);
      }
    };
    loadModelInfo(id);
  }, [id]);

  return model;
};

const ContactButton = ({ phoneNumber }, { guardianPhoneNumber }) => {
  const hasNoPhoneNumber = () => {
    return !(guardianPhoneNumber || phoneNumber);
  };

  const callPhoneNumber = () => {
    window.location = `tel:${guardianPhoneNumber || phoneNumber}`;
  };

  return (
    <div className={classes.ContactButton}>
      <Button
        disabled={hasNoPhoneNumber()}
        className={classes.Button}
        clicked={callPhoneNumber}
      >
        Contatar
      </Button>
    </div>
  );
};

const MainInfo = ({ model }) => {
  const birthday = model.birthday;
  const genderExpression = model.genderExpression;
  const name = model.name;

  return (
    <>
      <div className={classes.MainInfos}>
        <Information testId="name" label="Nome:" strong>
          {name}
        </Information>
        {birthday && (
          <Information label="Idade:" strong>
            {calculateAge(birthday)}
          </Information>
        )}
      </div>
      <Information label="Expressão de gênero:" strong>
        {GENDER_EXPRESSION[genderExpression]}
      </Information>
    </>
  );
};

const SuccessMessage = ({ location }) => {
  return location.state && location.state.registrationSuccessful ? (
    <Alert severity="success" className={classes.SucessMessage}>
      Os dados foram salvos com sucesso!
    </Alert>
  ) : null;
};

const Notes = ({ notes }) => {
  return notes ? (
    <>
      <h1>OBSERVAÇÕES:</h1>
      <span>{notes}</span>
    </>
  ) : null;
};
