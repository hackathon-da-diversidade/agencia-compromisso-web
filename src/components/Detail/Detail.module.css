h1 {
  font-size: 16px;
  font-weight: bold;
  margin-top: 2em;
}

.MainInfos {
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
}

.Details {
  display: inline-block;
  width: 90%;
  max-width: 400px;
  text-align: left;
}

.SucessMessage {
  margin-bottom: 30px;
}

.ContactButton {
  display: flex;
  justify-content: center;
}

@media only screen and (min-width: 600px) {
  .ContactButton {
    display: none;
  }
}
