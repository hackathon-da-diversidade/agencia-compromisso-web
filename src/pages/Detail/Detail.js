import React, { useEffect, useState } from 'react';
import Alert from '@material-ui/lab/Alert';
import Header from 'components/Header/Header';
import MainInformation from './components/MainInformation/MainInformation';
import SuccessMessage from './components/SuccessMessage/SuccessMessage';
import PersonalInformation from './components/PersonalInformation/PersonalInformation';
import MeasuresInformation from './components/MeasuresInformation/MeasuresInformation';
import SocialInformation from './components/SocialInformation/SocialInformation';
import Notes from './components/Notes/Notes';
import ContactButton from './components/ContactButton/ContactButton';

import candidateAPI from 'api/candidateAPI';
import classes from './Detail.module.css';
import { withRouter } from 'react-router-dom';

const useCandidate = (id) => {
  const [candidate, setCandidate] = useState({
    sizes: {},
    socialInformation: {},
  });

  useEffect(() => {
    const loadCandidateInfo = async (id) => {
      try {
        const { data } = await candidateAPI.get(id);
        setCandidate(data);
      } catch (error) {
        console.log(error);
      }
    };

    loadCandidateInfo(id);
  }, [id]);

  return candidate;
};

const Detail = ({ match, location }) => {
  try {
    const candidate = useCandidate(match.params.id);
    return (
      <div>
        <Header path="/lista" />
        {candidate.name ? (
          <div className={classes.Details}>
            <SuccessMessage location={location} />
            <MainInformation candidate={candidate} />
            <PersonalInformation data={candidate} />
            <MeasuresInformation data={candidate.sizes} />
            <SocialInformation data={candidate.socialInformation} />
            <Notes>{candidate.notes}</Notes>
            <ContactButton
              phoneNumber={candidate.phoneNumber}
              guardianPhoneNumber={candidate.guardianPhoneNumber}
            />
          </div>
        ) : null}
      </div>
    );
  } catch (error) {
    console.log(error);
    return (
      <Alert severity="warning">Não foi possível carregar o perfil.</Alert>
    );
  }
};

export default withRouter(Detail);
