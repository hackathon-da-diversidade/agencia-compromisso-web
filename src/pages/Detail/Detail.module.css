h1 {
  font-size: 16px;
  font-weight: bold;
  margin-top: 2em;
}

.ImagesWrapper img {
  width: 100%;
  border-radius: 4px;
}

.ImagesPaper {
  padding: 1%;
}

.ImagesWrapper {
  display: flex;
  justify-content: center;
  margin-top: 5%;
  margin-bottom: 5%;
}

.Alert {
  margin-top: 5%;
  margin-bottom: 5%;
}

.Details {
  display: inline-block;
  width: 90%;
  text-align: left;
}

@media (min-width: 768px) {
  .Details {
    width: 75%;
  }
}

@media (min-width: 992px) {
  .Details {
    width: 50%;
  }
}

@media (min-width: 1200px) {
  .Details {
    width: 25%;
  }
}
