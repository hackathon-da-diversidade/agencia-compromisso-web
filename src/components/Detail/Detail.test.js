import React from 'react';
import { render, wait } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';

import Detail from './Detail';

const detailWithAPIMock = data => {
  jest.mock('../../api/fitModelAPI', () => ({
    get: () =>
      new Promise((resolve, reject) => {
        const data = { data };
        resolve(data);
      }),
  }));

  const match = { params: { id: '5e3351db14c1174edbc11ac3' } };

  const location = {};

  return (
    <BrowserRouter>
      <Detail match={match} location={location} />
    </BrowserRouter>
  );
};

it('match snapshot', async () => {
  const data = {
    id: '5e3351db14c1174edbc11ac3',
    name: 'Maria',
    birthday: '21/10/2010',
    availability: 'MORNING',
    phoneNumber: '(41)123123333',
    projects: 'Projeto 1',
    address: null,
    genderExpression: 'FEMALE',
    education: null,
    guardianName: 'Pai da Maria',
    guardianPhoneNumber: '(41)231231231',
    sizes: {
      totalBustCircumference: 100.0,
      totalWaistCircumference: 100.0,
      totalHipCircumference: 100.0,
      height: 100.0,
    },
    socialInformation: null,
    identifyAsLGBTQIA: false,
  };

  const { asFragment, getByText } = render(detailWithAPIMock(data));
  await wait(() => getByText('Maria'));

  expect(asFragment()).toMatchSnapshot();
});

it('renders all main information', async () => {
  const data = {
    id: '5e3351db14c1174edbc11ac3',
    name: 'Maria',
    birthday: '21/10/2010',
    genderExpression: 'FEMALE',
  };

  jest.mock('../../api/fitModelAPI', () => ({
    get: () =>
      new Promise((resolve, reject) => {
        const data = { data };
        resolve(data);
      }),
  }));

  const match = { params: { id: '5e3351db14c1174edbc11ac3' } };

  const location = {};

  const { asFragment, getByText } = render(
    <BrowserRouter>
      <Detail match={match} location={location} />
    </BrowserRouter>
  );
  await wait(() => getByText('Maria'));

  expect(getByText(data.name)).toBeVisible();
  expect(getByText('Idade:')).toBeVisible();
  expect(getByText('Expressão de gênero:')).toBeVisible();
  expect(getByText('Feminina')).toBeVisible();
});

it('renders all main information', async () => {
  const data = {
    id: '5e3351db14c1174edbc11ac3',
    name: 'Maria',
    birthday: '21/10/2010',
    genderExpression: 'FEMALE',
  };

  jest.mock('../../api/fitModelAPI', () => ({
    get: () =>
      new Promise((resolve, reject) => {
        const data = { data };
        resolve(data);
      }),
  }));

  const match = { params: { id: '5e3351db14c1174edbc11ac3' } };

  const location = {};

  const { asFragment, getByText } = render(
    <BrowserRouter>
      <Detail match={match} location={location} />
    </BrowserRouter>
  );
  await wait(() => getByText('Maria'));

  expect(getByText(data.name)).toBeVisible();
  expect(getByText('Idade:')).toBeVisible();
  expect(getByText('Expressão de gênero:')).toBeVisible();
  expect(getByText('Feminina')).toBeVisible();
});

// it('não renderiza label se não tem informação', () => {
//   fitModelAPI.get(id);
//   // const { asFragment } = render(<Detail />);
//   // expect(asFragment()).toMatchSnapshot();
// });

// it('não renderiza botão de contatar se não tem telefone', () => {
//   fitModelAPI.get(id);
//   // const { asFragment } = render(<Detail />);
//   // expect(asFragment()).toMatchSnapshot();
// });

// it('renderiza botão de contatar se tem telefone', () => {
//   fitModelAPI.get(id);
//   // const { asFragment } = render(<Detail />);
//   // expect(asFragment()).toMatchSnapshot();
// });

// it('mostra alerta de sucesso caso registro do modelo seja salvo (registrationSuccessful) ', () => {
//   fitModelAPI.get(id);
// });
