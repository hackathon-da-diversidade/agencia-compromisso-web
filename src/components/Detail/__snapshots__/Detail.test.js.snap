// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="Header"
    >
      <a
        class="HeaderBack"
        href="/lista"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </svg>
      </a>
      <img
        alt="ícone estilizado com o perfil de uma mulher com uma fita métrica no black power"
        class="HeaderIcon"
        src="icon.png"
      />
      <img
        alt="banner da Agência Compromisso"
        class="HeaderBanner"
        src="banner.png"
      />
      <h2
        class="HeaderTitle"
      />
    </div>
    <div
      class="Details"
    >
      <div
        class="MainInfos"
      >
        <div
          class="Information"
        >
          <strong>
            Nome:
          </strong>
          <span>
            Maria
          </span>
        </div>
        <div
          class="Information"
        >
          <strong>
            Idade:
          </strong>
          <span>
            9
          </span>
        </div>
      </div>
      <div
        class="Information"
      >
        <strong>
          Expressão de gênero:
        </strong>
        <span>
          Feminina
        </span>
      </div>
      <h1>
        MEDIDAS:
      </h1>
      <div
        class="Information"
      >
        Circunferência total do busto (cm):
        <span>
          100
        </span>
      </div>
      <div
        class="Information"
      >
        Circunferência total da cintura alta(cm):
        <span>
          100
        </span>
      </div>
      <div
        class="Information"
      >
        Circunferência total do quadril (cm):
        <span>
          100
        </span>
      </div>
      <div
        class="Information"
      >
        Altura (cm):
        <span>
          100
        </span>
      </div>
      <h1>
        PESSOAL:
      </h1>
      <div
        class="Information"
      >
        Data de nascimento:
        <span>
          21/10/2010
        </span>
      </div>
      <div
        class="Information"
      >
        Disponibilidade:
        <span>
          Manhã
        </span>
      </div>
      <div
        class="Information"
      >
        Projetos:
        <span>
          Projeto 1
        </span>
      </div>
      <div
        class="Information"
      >
        Telefone:
        <span>
          (41)123123333
        </span>
      </div>
      <div
        class="Information"
      >
        Pertence à comunidade LGBTQIA+:
        <span>
          Não
        </span>
      </div>
      <div
        class="ContactButton"
      >
        <button
          class="Button"
        >
          Contatar
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
